WIP: use user interactions to whitelist hosts.

cowlicks opened the pull request:
This implements something like what Cliqz does, as described by this [this issue](https://github.com/cliqz-oss/browser-core/issues/58).

We create an ephemeral whitelist of domains from user interactions, then check this whitelist when we call `isThirdParty`

Todo:
* we should probably attach the whitelist to `Tabs` instead of each `Tab`.
* add tests
* actually find a domain where this fixes an issue, and test it
* make the whitelist a FIFO cache
* cache duplicate user interactions in the contentscript so we don't send too many messages

Related to #72 and #4

jotake commented:
I am involved  with some files than need some change also.
I I will get some important  information from your work.
Have a nice day
Good work

jeroen7s commented:
any updates on this?

cowlicks commented:
@jeroen7s hi! I just rebased this on to master. I think I had intended to merge this after I added selenium tests, and could add tests for this PR. Adding selenium tests was done in #103 and #125. If you'd like to help, I need to go through that todo checklist at the top of this PR, and add tests.
